modules/cli/src/main/scala/scala/cli/commands/util/FmtUtil.scala

```diff
@@ -0,0 +1,194 @@
+        metaconfig.generic.deriveDecoder[ScalafmtMetaconfig](default)
+    }
+
+    val confName = ".scalafmt.conf"
+    val pathMaybe = {
+      logger.debug(s"Checking for $confName in cwd.")
+      val confInCwd = workspace / confName
+      if (os.exists(confInCwd)) Some(confInCwd)
+      else {
+        logger.debug(s"Checking for $confName in git root.")
+        val gitRootMaybe       = getGitRoot(workspace, logger)
+        val confInGitRootMaybe = gitRootMaybe.map(os.Path(_) / confName)
+        confInGitRootMaybe.find(os.exists(_))
+      }
+    }
+
+    val confContentMaybe = pathMaybe.flatMap { path =>
+      val either = metaconfig.Hocon.parseInput[ScalafmtMetaconfig](
+        metaconfig.Input.File(path.toNIO)
+      ).toEither
+      either.left.foreach(confErr => logger.log(confErr.toString()))
+      either.toOption
+    }
+    val versionMaybe = confContentMaybe.collect {
+      case conf if conf.version.trim.nonEmpty => conf.version
+    }
+    val dialectMaybe = confContentMaybe.collect {
+      case conf if conf.runner.dialect.trim.nonEmpty => conf.runner.dialect
+    }
+    (versionMaybe, dialectMaybe, pathMaybe)
+  }
+
+  // Based on https://github.com/scalameta/metals/blob/main/metals/src/main/scala/scala/meta/internal/metals/ScalafmtDialect.scala
+  sealed abstract class ScalafmtDialect(val value: String)
+  object ScalafmtDialect {
+    case object Scala3          extends ScalafmtDialect("scala3")
+    case object Scala213        extends ScalafmtDialect("scala213")
+    case object Scala213Source3 extends ScalafmtDialect("scala213source3")
+    case object Scala212        extends ScalafmtDialect("scala212")
+    case object Scala212Source3 extends ScalafmtDialect("scala212source3")
+    case object Scala211        extends ScalafmtDialect("scala211")
+
+    implicit val ord: Ordering[ScalafmtDialect] = new Ordering[ScalafmtDialect] {
+
+      override def compare(x: ScalafmtDialect, y: ScalafmtDialect): Int =
+        prio(x) - prio(y)
+
+      private def prio(d: ScalafmtDialect): Int = d match {
+        case Scala211        => 1
+        case Scala212        => 2
+        case Scala212Source3 => 3
+        case Scala213        => 4
+        case Scala213Source3 => 5
+        case Scala3          => 6
+      }
+    }
+
+    def fromString(v: String): Option[ScalafmtDialect] = v.toLowerCase match {
+      case "default"         => Some(Scala213)
+      case "scala211"        => Some(Scala211)
+      case "scala212"        => Some(Scala212)
+      case "scala212source3" => Some(Scala212Source3)
+      case "scala213"        => Some(Scala213)
+      case "scala213source3" => Some(Scala213Source3)
+      case "scala3"          => Some(Scala3)
+      case _                 => None
+    }
+  }
+
+  // Based on https://github.com/scalameta/metals/blob/main/metals/src/main/scala/scala/meta/internal/metals/ScalafmtConfig.scala
+  def updateScalafmtConfig(
```

Comment:
the `update` in the function name is confusing IMO, as it suggests there's a side-effect, while in fact, the function is pure.

hard to name it well, maybe smth like:
```suggestion
  def scalafmtConfigWithFields(
```

some scaladoc might help, maybe